common/str.cpp

```diff
@@ -444,6 +444,29 @@ uint String::hash() const {
 	return hashit(c_str());
 }
 
+void String::replace(uint pos, uint count, const String& str) {
+	replace(pos, count, str._str, 0, count);
+}
+
+void String::replace(uint pos, uint count, const char* str) {
+	replace(pos, count, str, 0, count);
+}
+
+void String::replace(iterator begin, iterator end, const String& str) {
```

Comment:
Maybe in this function it would be good to assert that str is long enough to avoid accessing characters beyond the end of the string.